instance-group-ips-build-task/instance-group-ips.ts

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Endpoint, getDefaultExecOptions} from 'common/exec-options';
+import * as task from 'vsts-task-lib/task';
+
+/**
+ * @fileoverview This is the implementation of the instance group ips task.
+ * @author jimwp@google.com (Jim Przybylinski)
+ */
+
+/**
+ * The path for retrieving the external IPs of a virtual machine.
+ * Copied from the gcloud default EXTERNAL_IP field.
+ */
+const externalIpResource =
+  'networkInterfaces[].accessConfigs[0].natIP.notnull().list()';
+/**
+ * The path for the name of a virtual machine.
+ */
+const instanceNameKey = 'instance.scope().segment(2)';
+
+/**
+ * The path for the zone of a virtual machine.
+ */
+const instanceZoneKey = 'instance.scope().segment(0)';
+
+/**
+ * Interface describing the inputs to {@link getInstanceGroupIps}
+ */
+export interface GetInstanceGroupIpsOptions {
+  endpoint: Endpoint;
+  locationScope: string;
+  location: string;
+  instanceGroupName: string;
+  separator?: string;
+  buildVariableName: string;
+  di?: {task: typeof task};
+}
+
+/**
+ * The implementation for finding an instance group and getting the external
+ * IPs of its member virtual machines.
+ */
+export async function getInstanceGroupIps(
+    {
+      endpoint,
```

Comment:
nit: This parameter name is easily confused with `Endpoint`.

Reply:
It is an instance of `Endpoint`.